llvm/lib/Transforms/Utils/LoopIterCountAssumptions.cpp

```diff
@@ -131,11 +131,12 @@ void insertMinIterAssumption(ICmpInst &LoopCmpInstr, BasicBlock &LoopHeader,
   // expression can be simplified in later passes
   SCEV::NoWrapFlags NWF = BunldedSCEV.AddRecExpr->getNoWrapFlags(
       SCEV::NoWrapFlags(/*Mask=*/SCEV::FlagNUW | SCEV::FlagNSW));
-  SCEVCommutativeExpr *AE =
-      dyn_cast<SCEVCommutativeExpr>(const_cast<SCEV *>(MinIterSCEV));
-  if (AE && NWF) {
-    AE->setNoWrapFlags(NWF);
-    MinIterSCEV = dyn_cast<const SCEV>(AE);
+  const SCEVCommutativeExpr *ConstCE =
+      dyn_cast<SCEVCommutativeExpr>(MinIterSCEV);
+  if (ConstCE && NWF) {
+    SCEVCommutativeExpr *CE = const_cast<SCEVCommutativeExpr *>(ConstCE);
+    CE->setNoWrapFlags(NWF);
```

Comment:
Hmm, that might be dangerous to const_cast an expression and change its properties? (It might be cached somewhere and re-used) Is there a way to get a non-const `SCEV` from `evaluateAtIteration()`?

Reply:
I unfortunately get a const SCEV expression from `evaluateAtIteration()`